src/Control/Monad/Throws.hs

```diff
@@ -0,0 +1,104 @@
+{-# language FunctionalDependencies #-}
+{-# OPTIONS_GHC -Wredundant-constraints #-}
+
+module Control.Monad.Throws where
+
+import Control.Monad.Catch (MonadThrow, MonadCatch, throwM, Exception)
+import Lib
+import GHC.Stack
+import GHC.TypeLits
+import qualified Control.Exception.Safe as Exception
+
+class (Monad m, Exception err) => MonadThrows err m | m -> err where
+    throwAllChecked :: (HasCallStack) => err -> m a
+
+throwChecked :: (MonadThrows err m, ExceptionSubtype err e) => e -> m a
+throwChecked = throwAllChecked . inject
+
+instance
+    ( Exception err, MonadThrow m
+    ) => MonadThrows err (CheckedT err m) where
+    throwAllChecked = throwUnchecked
```

Comment:
And maybe surprisingly enough - this instance is all that you need to make the types work for `try`!